src/modeler/AutoRest.Swagger/OperationBuilder.cs

```diff
@@ -222,10 +215,37 @@ private List<Stack<IModelType>> BuildResponses(Method method, CompositeType head
                     }
                 }
             }
-
+            RemoveResponsesWithDefaultType(method, typesList);
             return typesList;
         }
 
+        private void RemoveResponsesWithDefaultType(Method method, List<Stack<IModelType>> typesList)
+        {
+            var errorModelType = method.DefaultResponse.Body as CompositeType;
+            if (errorModelType == null || method.Responses == null) return;
+            var keys = method.Responses.Keys;
+            var removeResponses = new List<HttpStatusCode>();
+            foreach (var key in keys)
```

Comment:
The response types for error conditions should be _related_  (ie, default uses `#/definitions/SomeError` and all the other error responses should use something that either returns that type, or has an allOf: "#/definitions/SomeError" in it's type declaration.)

That way we can have very specific types for errors, and makes it very clear what's an acceptable error.

Of course that means we have to do three other things:
- make sure that use of the `allOf` doesn't force type-inheritance, it's possible that we should be able to have two error types that bubble up to the  same swagger model type (that doesn't even have any properties), but still generate individual classes without a hierarchy.

- change the generator to throw the correct exception type, rather than living with the default class

- and add method documentation for known possible exceptions